docs/lib/restapi/fragments/native_common/authz/common.md

```diff
@@ -0,0 +1,124 @@
+
+## IAM Authorization
+
+By default, the API will be using IAM authorization and the requests will be signed for you automatically. IAM authorization has two modes of authorization, an unauthenticated role and an authenticated role. When the user has not signed in through Amplify.Auth.signIn then the assumed role will be unauth. Once the user has signed in, then the role with be auth.
+
+When creating the REST API, you assigned the permissions for these roles when you were asked if you wanted to restrict access. If no, then unauth role will have access to the API. If yes, you would have configured more fine grain access to your API.
+
+### Unauthenticated Requests
+
+You can use the API category to access API Gateway endpoints that don't require authentication. In this case, you need to allow unauthenticated identities in your Amazon Cognito Identity Pool settings. For more information, please visit [Amazon Cognito Developer Documentation](https://docs.aws.amazon.com/cognito/latest/developerguide/identity-pools.html#enable-or-disable-unauthenticated-identities).
+
+Your requests will be automatically have the IAM credentials when your API is set up to use IAM as the authorization mode. For example, you can check this by looking at what authorization type is configured for the API
+
+```json
+{
+    "awsAPIPlugin": {
+        "<YOUR-RESTENDPOINT-NAME>": {
+            "endpointType": "REST",
+            "endpoint": "YOUR-REST-ENDPOINT",
+            "region": "us-west-2",
+            "authorizationType": "IAM"
+        }
+    }
+}
+```
+
+## API Key
+
+If you want to configure the REST API to be a public API, you can set an API key on the API Gateway and update your configuration to use the API key. You should have the following configuration in your `amplifyconfiguration.json` file containing the `API_KEY` authorization mode and the value of the API key to be used when creating requests.
+
+```json
+{
+    "awsAPIPlugin": {
+        "<YOUR-RESTENDPOINT-NAME>": {
+            "endpointType": "REST",
+            "endpoint": "YOUR-REST-ENDPOINT",
+            "region": "us-west-2",
+            "authorizationType": "API_KEY",
+            "API_KEY": "[API_KEY]"
+        }
+    }
+}
+```
+
+
+## Cognito User pool authorization
+
+If you set up the API Gateway with a custom authorization with a Cognito User pool, then you can set the `amplifyconfiguration.json` to use `COGNITO_USER_POOL`. 
+
+```json
+{
+    "awsAPIPlugin": {
+        "<YOUR-RESTENDPOINT-NAME>": {
+            "endpointType": "REST",
+            "endpoint": "YOUR-REST-ENDPOINT",
+            "region": "us-west-2",
+            "authorizationType": "COGNITO_USER_POOL"
+        }
+    }
+}
+```
+
+Also in `amplifyconfiguration.json` file, under `awsCognitoAuthPlugin` will contain the corresponding Cognito configuration values that refer to the Cognito User Pool you want to use.
+```json
+{
```

Comment:
In this snippet, I suggest to include more context. I would start _all_ of your code snippets with `"awsAPIPlugin"`, so that the reader can compare them to one another and clearly distinguish the semantic changes _only for_ auth.